prow/cmd/validation-webhook/main_test.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"testing"
+)
+
+type SecretStore struct {
+	store map[string]string
+	dirName string
+} 
+
+type FakeClient struct {
+	project SecretStore
+}
+
+type FakeClientInterface interface {
+	CreateSecret(ctx context.Context, secretID string)
+	AddSecretVersion(ctx context.Context, secretName string) 
+	GetSecretValue(ctx context.Context, secretName string, versionName string) (string, error)
+}
+
+
+func (f *FakeClient) CreateSecret(ctx context.Context, secretID string) {
+	f.project.dirName = secretID
+}
+
+func (f *FakeClient) AddSecretVersion(ctx context.Context, secretName string, payload []byte) {
+	f.project.store[secretName] = string(payload)
+}
+
+func (f *FakeClient) GetSecretValue(ctx context.Context, secretName string, versionName string) (string, error) {
+	if f.project.dirName != secretName {
+		return "", errors.New("Secret was not created!")
+	} else {
+		if val, ok := f.project.store[versionName]; ok {
+			return val, nil
+		} else {
+			err := fmt.Sprintf("Secret with name %s was never added", versionName)
+			return "", errors.New(err)
+		}
+	}
+}
+
+func TestSecretsExist(t *testing.T) {
+	cert, certPrivKey, certCaBundle := genCert()
```

Comment:
After this line we don't actually call any non-test code and we don't check the values returned on this line so we aren't actually validating anything besides the fake client.
Rather than defining a `FakeClientInterface` here, define and use a `ClientInterface` in your main code and then pass the `FakeClient` rather than a real client to test functions like `getSecrets` and `createGCPSecret`.

The same applies to `TestSecretsDontExist`.